### PR TITLE
Researchers

### PR DESCRIPTION
**Pull request by jmbarne3**
Enhancements:
* Added a `Researcher` model which ties an ORCID ID back to a `teledata.Staff` object. We may end up adding additional metadata to this, but for now I think this is sufficient since the end goal is to be able to attach research (publications, projects, etc) back to UCF faculty, who should be listed in our teledata.
* Added an import process that matches the given and family names in ORCID to the first and last names in our teledata. This matching process may need to be improved upon in the future, but I'm going to hold off on trying to do anything until we have an idea of how many faculty members are using the system, so I have a number to benchmark against.

### CODE REVIEW

File: research/management/commands/orcid-researcher-import.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        """
+        Private helper method for retrieving ORCID records
+        """
+        if page_num != 0:
+            params['start'] = page_num * 100
+
+        response = requests.get(request_url, params, headers=headers)
+
+        if response.status_code != response.ok:
+            CommandError("There was an error retriving the results from ORCID")
```
Comment (cjg89) [inline research/management/commands/orcid-researcher-import.py:208]: `CommandError` needs to be `raise`'d

File: research/management/commands/orcid-researcher-import.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+            data = self.__request_records(0, request_url, {}, headers)
+
+            try:
+                first_name = data['person']['name']['given-names']['value']
+                last_name = data['person']['name']['family-name']['value']
+            except KeyError:
+                self.stdout.write("Unable to get first or last name from record.\n")
+                self.error += 1
+                continue
+            except TypeError:
```
Comment (cjg89) [inline research/management/commands/orcid-researcher-import.py:169]: Unique error messages for these particular types of errors could be helpful--alternatively, we could combine the two with something like `except (KeyError, TypeError):`

**Review by cjg89 — APPROVED** [review]

File: research/management/commands/orcid-researcher-import.py (hunk 1 above)
Comment (jmbarne3) [reply research/management/commands/orcid-researcher-import.py]: Also, `reponse.ok` returns a boolean value indicating whether or not the response returned an "ok" response. So this if statement was totally wrong. Updated.

File: research/management/commands/orcid-researcher-import.py (hunk 2 above)
Comment (jmbarne3) [reply research/management/commands/orcid-researcher-import.py]: Combined the two exceptions. I don't think we really care which one is thrown. In either case, there isn't a first or last name for us to process.

**Review by cadie — APPROVED** [review]

**Review by rjucf — APPROVED** [review]

**Pull request merged**